tests/e2e/test_e2e.py

```diff
@@ -421,14 +421,14 @@ def test_e2e(caplog):
     ]
     labeler.update(split=0, lfs=[stg_temp_lfs_2, ce_v_max_lfs], parallelism=PARALLEL)
     assert session.query(Label).count() == 6478
-    assert session.query(LabelKey).count() == 16
+    assert session.query(LabelKey).count() == 19
```

Comment:
Why there are 3 more LabelKey here?

Reply:
The total number of unique LFs, `len(set(stg_temp_lfs + ce_v_max_lfs + stg_temp_lfs_2))`, is 20, while `session.query(LabelKey).all()` does not include `LF_temp_outside_table`, as a result `session.query(LabelKey).count()` is 19.

Like `LF_operating_row` returns `0` (ABSTAIN) for all candidates when `ABSTAIN==0`, `LF_temp_outside_table` returns `0` (FALSE) for all of them when `ABSTAIN==-1 (FALSE==0)`.

Because `Labeler` skips records with `value==0`, no labels for `LF_operating_row` is recorded.

Reply:
The Snorkel team standardized their coding strategy and API as follows:
> label matrix is dense outside of compute modules
> Internal computation for LF analysis and label model is with shifted sparse matrix: csr_matrix(L + 1)

https://github.com/snorkel-team/snorkel/pull/1309#issue-300125859

Reply:
Using -1 as ABSTAIN is not a computation efficient way since most of the values are ABSTAIN.

Reply:
Good catch!

Reply:
One more thought here, the total number of LabelKey should be 20 since the 0 means negative (no longer means abstain) and labeler should not ignore it.

Reply:
Good catch!
To avoid any confusion and computational inefficiency, how about stick with snorkel's policy?
> ABSTAIN is now represented by None (LFs), -1 (label matrix), or 0 (LFAnalysis, except for when we use L and then it's -1).
https://github.com/snorkel-team/snorkel/pull/1309#pullrequestreview-265185691

So -1 for user-facing labels and 0 for internal computation and database?